packages/transport-webrtc/test/certificate.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { Crypto } from '@peculiar/webcrypto'
+import { expect } from 'aegir/utils/chai.js'
+import sinon from 'sinon'
+import { WebRTCDirectListener, type WebRTCDirectListenerInit } from '../src/private-to-public/listener.js'
+import { type WebRTCDirectTransportComponents } from '../src/private-to-public/transport.js'
+import { generateTransportCertificate } from '../src/private-to-public/utils/generate-certificates.js'
+import type { TransportCertificate } from '../src/index.js'
+
+const crypto = new Crypto()
+
+describe('WebRTCDirectListener', () => {
+  let listener: WebRTCDirectListener
+  let components: WebRTCDirectTransportComponents
+  let init: WebRTCDirectListenerInit
+
+  beforeEach(() => {
+    components = {
+      peerId: { toB58String: () => 'QmPeerId' } as any,
+      privateKey: {} as any,
+      logger: {
+        forComponent: () => ({
+          trace: () => {},
+          error: () => {}
+        })
+      } as any,
+      transportManager: {} as any
```

Comment:
You can use `stubInterface<TransportManager>()` from `sinon-ts` here to avoid casting to `any`.